### ISSUE TITLE
hyper parameter tuning after each epoch

### ISSUE THREAD

**Issue by Mayuresh999** [ultralytics/ultralytics#3650]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Does yolov8 do hyper-parameter tuning after each epoch ? 
The model is tested after each epoch on validation set. Is there any input given from validation metrics that will change training process under the hood? 
Will the model be better if input from validation is given to update hyper parameters ? @glenn-jocher @AyushExel

**Comment by AyushExel** [comment]
No hyperparameter tuning doesn't happen after each epoch currently. Also, I haven't heard about it anywhere. Generally, all parameters are scaled(tuned) based on number of epochs so changing them after each epoch would not be optimal. There are some specialised genetic algorithms( population based training/tuning) that use data from other trainings to update the params in real time but there's a lot happening under the hood there. You might want to take a look
